code/modules/atmospherics/environmental/LINDA_fire.dm

```diff
@@ -305,4 +359,99 @@
 /obj/effect/hotspot/singularity_pull()
 	return
 
+/datum/looping_sound/fire
+	mid_sounds = list('sound/effects/fireclip1.ogg' = 1, 'sound/effects/fireclip2.ogg' = 1, 'sound/effects/fireclip3.ogg' = 1, 'sound/effects/fireclip4.ogg' = 1,
+	'sound/effects/fireclip5.ogg' = 1, 'sound/effects/fireclip6.ogg' = 1, 'sound/effects/fireclip7.ogg' = 1)
+	volume = 100
+	mid_length = 2 SECONDS
+	falloff_distance = 1
+///handle the grouping of hotspot and then determining an average center to play sound in
+/datum/hot_group
+	var/list/obj/effect/hotspot/spot_list = list()
+	///the sound center turf which the looping sound will play
+	var/turf/open/current_sound_loc
+	var/datum/looping_sound/fire/sound
+	var/tiles_limit = 80 // arbitrary limit so we dont have one giant group
+	///these lists and average var are to find the average center of a group
+	var/list/x_coord = list()
+	var/list/y_coord = list()
+	var/list/z_coord = list()
+	var/average_x
+	var/average_y
+	var/average_Z
+	///the range for the sound to drop off based on the size of the group
+	var/drop_off_dist
+	COOLDOWN_DECLARE(update_sound_center)
+
+
+/datum/hot_group/Destroy()
+	. = ..()
+	current_sound_loc = null
+	spot_list = null
+	qdel(sound)
+
+/datum/hot_group/proc/remove_from_group(obj/effect/hotspot/target)
+	spot_list -= target
+	var/turf/open/target_turf = target.loc
+	x_coord -= target_turf.x
+	y_coord -= target_turf.y
+	if(!length(spot_list))
+		qdel(src)
+		return
+
+/datum/hot_group/proc/add_to_group(obj/effect/hotspot/target)
+	spot_list += target
+	target.our_hot_group = src
+	var/turf/open/target_turf = target.loc
+	x_coord += target_turf.x
+	y_coord += target_turf.y
+	z_coord += target_turf.z
+	if(COOLDOWN_FINISHED(src, update_sound_center) && length(spot_list) > 2)//arbitrary size to start playing the sound
```

Comment:
should make your arbitrary size a define to make said arbitraryness and universalness clear